include/rotatingMHD/convection_diffusion/assembly_data.h

```diff
@@ -0,0 +1,148 @@
+#ifndef INCLUDE_ROTATINGMHD_CONVECTION_DIFFUSION_ASSEMBLY_DATA_H_
+#define INCLUDE_ROTATINGMHD_CONVECTION_DIFFUSION_ASSEMBLY_DATA_H_
+
+#include <deal.II/base/tensor.h>
+
+#include <rotatingMHD/assembly_data_base.h>
+
+#include <vector>
+
+namespace RMHD
+{
+
+namespace AssemblyData
+{
+
+namespace HeatEquation
+{
+
+namespace ConstantMatrices
+{
+
+using Copy = AssemblyData::Generic::Matrix::MassStiffnessCopy;
+
+template<int dim>
+struct Scratch : Generic::Matrix::Scratch<dim>
+{
+  Scratch(const Mapping<dim>        &mapping,
+          const Quadrature<dim>     &quadrature_formula,
+          const FiniteElement<dim>  &fe,
+          const UpdateFlags         update_flags);
+
+  Scratch(const Scratch<dim>    &data);
+
+  std::vector<double>         phi;
+
+  std::vector<Tensor<1, dim>> grad_phi;
+};
+
+} // namespace ConstantMatrices
+
+namespace AdvectionMatrix
+{
+
+using Copy = Generic::Matrix::Copy;
+
+template <int dim>
+struct Scratch : ScratchBase<dim>
+{
+  Scratch(const Mapping<dim>        &mapping,
+          const Quadrature<dim>     &quadrature_formula,
+          const FiniteElement<dim>  &temperature_fe,
+          const UpdateFlags         temperature_update_flags,
+          const FiniteElement<dim>  &velocity_fe,
+          const UpdateFlags         velocity_update_flags);
+
+  Scratch(const Scratch<dim>    &data);
+
+  FEValues<dim>               temperature_fe_values;
+
+  FEValues<dim>               velocity_fe_values;
+
+  std::vector<Tensor<1,dim>>  velocity_values;
+
+  std::vector<Tensor<1,dim>>  old_velocity_values;
+
+  std::vector<Tensor<1,dim>>  old_old_velocity_values;
+
+  std::vector<double>         phi;
+
+  std::vector<Tensor<1,dim>>  grad_phi;
+};
+
+} // namespace AdvectionMatrix
+
+namespace RightHandSide
+{
+
+using Copy = Generic::RightHandSide::Copy;
+
+template <int dim>
+struct CDScratch : ScratchBase<dim>
```

Comment:
I am not a fan of acronyms in the name of classes/structs

Reply:
I will add some documentation...